website/mailchimp_utils.py

```diff
@@ -27,20 +28,28 @@ def subscribe(list_name, user_id):
     user = User.load(user_id)
     m = get_mailchimp_api()
     list_id = get_list_id_from_name(list_name=list_name)
-    m.lists.subscribe(id=list_id,
-                      email={'email': user.username},
-                      merge_vars={'fname': user.given_name,
-                                  'lname': user.family_name},
-                      double_optin=False,
-                      update_existing=True)
 
-    # Update mailing_list user field
     if user.mailing_lists is None:
         user.mailing_lists = {}
         user.save()
 
-    user.mailing_lists[list_name] = True
-    user.save()
+    try:
+        m.lists.subscribe(id=list_id,
+                          email={'email': user.username},
+                          merge_vars={'fname': user.given_name,
+                                      'lname': user.family_name},
+                          double_optin=False,
+                          update_existing=True)
+
+    except mailchimp.ValidationError as error:
+        sentry.log_exception()
+        sentry.log_message(error.message)
+        user.mailing_lists[list_name] = False
+        user.save()
+
+    else:
+        user.mailing_lists[list_name] = True
+        user.save()
```

Comment:
Made a minor change during merge (see ebfc8a7): to prevent multiple save calls, moved `user.save()` to the `finally` block.